### PR TITLE
chore!: declare outputs for aadApp/create in templates

### PR DESCRIPTION
**Pull request by blackchoey**
1. declare outputs for aadApp/create in templates
2. mark it as breaking change to bump the major version of our templates

BREAKING CHANGE: the `writeToEnvironmentFile` property will be required in the future, and there will also be other breaking changes recently. So mark this PR as breaking change to bump the major version of our templates.

### CODE REVIEW

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl [hunk 1]
```diff
@@ -5,15 +5,15 @@ version: 1.0.0
...
+      name: {{appName}}-aad # Note: when you run aadApp/update, the AAD app name will be updated based on the definition of manifest. If you don't want to change the name, ensure the name in AAD manifest is same with the name defined here.
       generateClientSecret: true # If the value is false, the action will not generate client secret for you
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # AAD_APP_CLIENT_ID: the client id of AAD app
-    # AAD_APP_CLIENT_SECRET: the client secret of AAD app
-    # AAD_APP_OBJECT_ID: the object id of AAD app
-    # AAD_APP_TENANT_ID: the tenant id of AAD app
-    # AAD_APP_OAUTH_AUTHORITY_HOST: the host of OAUTH authority of AAD app
-    # AAD_APP_OAUTH_AUTHORITY: the OAUTH authority of AAD app
+    writeToEnvironmentFile: # Writes the information of created resources to specified environment variable in the environment file
```
Comment (blackchoey) [inline templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl:10]: @sffamily can you help review the updated wording in this file? The other files have the same change so you only need to review this file and I will apply your comments to all the files.

**Review by swatDong — DISMISSED** [review]

File: templates/scenarios/js/m365-tab/teamsapp.local.yml.tpl [hunk 2]
```diff
@@ -5,15 +5,15 @@ version: 1.0.0
 provision:
   - uses: aadApp/create # Creates a new AAD app to authenticate users if AAD_APP_CLIENT_ID environment variable is empty
```
Comment (xzf0587) [inline templates/scenarios/js/m365-tab/teamsapp.local.yml.tpl:6]: should we use aadApp/createIfNotExists?

Comment (blackchoey) [reply templates/scenarios/js/m365-tab/teamsapp.local.yml.tpl:6]: after discussed with Tim/John, will keep using `aadApp/create`

**Review by xzf0587 — DISMISSED** [review]

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl [hunk 3]
```diff
@@ -5,15 +5,15 @@ version: 1.0.0
 provision:
   - uses: aadApp/create # Creates a new AAD app to authenticate users if AAD_APP_CLIENT_ID environment variable is empty
     with:
-      name: {{appName}}-aad # Note: when you run configure/aadApp, the AAD app name will be updated based on the definition of manifest. If you don't want to change the name, ensure the name in AAD manifest is same with the name defined here.
+      name: {{appName}}-aad # Note: when you run aadApp/update, the AAD app name will be updated based on the definition of manifest. If you don't want to change the name, ensure the name in AAD manifest is same with the name defined here.
```
Comment (sffamily) [inline templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl:8]: of manifest -> in manifest
ensure -> make sure
same -> the same

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl (hunk 1 above)
Comment (sffamily) [inline templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl:10]: Writes -> Write

Comment (sffamily) [reply templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl:10]: Write the information of created resources into environment file for the specified environment variable(s).

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl (hunk 3 above)
Comment (blackchoey) [reply templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl]: thanks, updated

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl (hunk 1 above)
Comment (blackchoey) [reply templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl]: thanks, updated

**Review by swatDong — APPROVED** [review]

File: templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl [hunk 4]
```diff
@@ -3,17 +3,17 @@
 version: 1.0.0
 
 provision:
-  - uses: aadApp/create # Creates a new AAD app to authenticate users if AAD_APP_CLIENT_ID environment variable is empty
+  - uses: aadApp/create # Creates a new AAD app to authenticate users if the environment variable that stores clientId is empty
```
Comment (blackchoey) [inline templates/scenarios/csharp/sso-tab/teamsapp.local.yml.tpl:6]: @sffamily can you help review update to this line?

The other content has already been reviewed and I have resolved your comments.

**Review by sffamily — DISMISSED** [review]

**Comment by blackchoey** [comment]
I also updated all the templates to use full name for AAD for the first occurrence in the file, based on comments in PR https://github.com/OfficeDev/TeamsFx-Samples/pull/731

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**